### PR TITLE
drivers: w1: updates to reflect acquistion of maxim by analog devices

### PR DESCRIPTION
**Pull request by str4t0m**
After the acquisition of Maxim Integrated the documentation of these devices has been moved to the analog.com website. Redirects exist, so they are not broken yet, but we should not rely on that forever.
Therefore the links in the drivers and the sensor sample are updated.

### CODE REVIEW

**Review by MaureenHelm — DISMISSED** [review]
Thanks!

File: drivers/sensor/maxim/ds18b20/ds18b20.c [hunk 1]
```diff
@@ -5,12 +5,10 @@
  */
 
 /**
- * Driver for DS18B20 1-Wire temperature sensors
- * A datasheet is available at:
- * https://datasheets.maximintegrated.com/en/ds/DS18B20.pdf
- *
- * Driver also support the older DS18S20 1-Wire temperature sensors.
+ * Driver for DS18B20 and DS18S20 1-Wire temperature sensors
+ * A datasheets is available at:
```
Comment (kartben) [inline drivers/sensor/maxim/ds18b20/ds18b20.c:9]:
```suggestion
 * A datasheet is available at:
```

**Review by kartben — CHANGES_REQUESTED** [review]
Nit: you may want to split URL updates and other Maxim->Analog renames in different commits, or just reword the single commit message so that it's better aligned with the nature of the update (see e.g. some C files where there's no URL being updated, contrary to what the commit message says)

Thanks for this!

File: drivers/sensor/maxim/ds18b20/ds18b20.c (hunk 1 above)
Comment (str4t0m) [reply drivers/sensor/maxim/ds18b20/ds18b20.c:9]: Good catch! As now both datasheets are listed below what do you think about:
` Datasheets for the compatible sensors are available at:`

**Comment by str4t0m** [comment]
> Nit: you may want to split URL updates and other Maxim->Analog renames in different commits, or just reword the single commit message so that it's better aligned with the nature of the update (see e.g. some C files where there's no URL being updated, contrary to what the commit message says)

I started to just update the links, then noticed a few additional things.

I've decided to just update the commit message and additionally split off the changes in the sample into a separate commit. I hope the description is now better reflecting the changes done.

**Review by ozersa — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**